packages/klighd-core/src/search-bar/searchbar-panel.tsx

```diff
@@ -0,0 +1,707 @@
+                                        block: 'nearest'
+                                    })
+                                }, 0)
+                            }
+                        }
+                    }
+                }, [this.textRes[index],
+                    isSelected ? h('div', {
+                        style: {
+                            fontSize: '11px',
+                            color: '#007acc',
+                            marginTop: '4px'
+                        }
+                    }, [this.decodeId(result.id)]) : null
+                ])
+            }))
+        ])
+    }   
+
+    /**
+     * Zoom in on a certain element.
+     * @param elementId id of the element to zoom in to. 
+     */
+    private panToElement(elementId: string): void {
+        const action: FitToScreenAction = {
+            kind: 'fit',
+            elementIds: [elementId],
+            animate: true,
+            padding: 20,
+            maxZoom: 2
+        }
+        this.actionDispatcher.dispatch(action)
+    }
+
+    /** Resets the UI by removing tooltips and the result list */
+    private resetUI(): void {
+        const input = document.getElementById('search') as HTMLInputElement | null
+        const tagInput = document.getElementById('tag-search') as HTMLInputElement | null
+
+        if (input) input.value = ''
+        if (tagInput) tagInput.value = ''
+        if (this.tooltipEl) this.tooltipEl.style.display = 'none'
+
+        this.searchResults = []
+        this.textRes = []
+        this.searched = false
+        this.selectedIndex = -1
+        this.update()
+        this.actionDispatcher.dispatch(ClearHighlightsAction.create())
+    }
+
+    /**
+     * When pressing the escape key, the search input resets.
+     * @param event keypress (esc key)
+     */
+    private handleEscapeKey = (event: KeyboardEvent) => {
+        if (event.key === 'Escape') {
+            const active = document.activeElement as HTMLElement | null
+            const input = document.getElementById('search') as HTMLInputElement | null
+            const tagInput = document.getElementById('tag-search') as HTMLInputElement | null
+
+            if (active === input && input) {
+                input.value = ''
+                input.focus()
+            } else if (active === tagInput && tagInput) {
+                tagInput.value = ''
+                tagInput.focus()
+            }
+            
+             this.handleInputChange()
+        }
+    }
+
+    /**
+     * Closes the search bar with a key combination
+     * @param event keycombination ctrl+x
+     */    
+    private handleExitKeycombo = (event: KeyboardEvent) => {
+        if ((event.metaKey || event.ctrlKey) && event.key === 'x') {
+            this.changeVisibility(false)
+            this.searched = false
+            this.tagInputVisible = false
+            this.actionDispatcher.dispatch(ToggleSearchBarAction.create(this, SearchBar.ID, 'hide'))
+        }
+    }
```

Comment:
most other tools close the search bar with the esc key and clear the input by just using usual ctrl+A del or ctrl+backspace combinations common in text fields.